cpp/src/slice2java/Gen.cpp

```diff
@@ -4243,7 +4243,8 @@ Slice::Gen::ProxyVisitor::visitInterfaceDefEnd(const InterfaceDefPtr& p)
     out << nl << "public static " << prxName
         << " createProxy(com.zeroc.Ice.Communicator communicator, String proxyString)";
     out << sb;
-    out << nl << "return uncheckedCast(communicator.stringToProxy(proxyString));";
+    out << nl << "var ref = communicator.getInstance().referenceFactory().create(proxyString, null);";
```

Comment:
I'd rather not call the very internal Reference factory from the generated code!

In C#, we create a base ObjectPrx and give it to the constructor. I would do the same here. It's slightly wasteful to create this temporary base ObjectPrx, however the speed of proxy creation doesn't matter. And it's now much faster since it no longer uses reflection!

Reply:
I've make the suggested change for consistency's sake,
but I feel like there must be a better way to do this without making an extra object here.